projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py

```diff
@@ -292,13 +308,15 @@ def on_close(self) -> None:
         """
         # stop serving this client
         self._message_queue.put_nowait(b"")
-        if isinstance(self.room, DocumentRoom) and self.room.clients == [self]:
-            # no client in this room after we disconnect
-            # keep the document for a while in case someone reconnects
-            self.log.info("Cleaning room: %s", self._room_id)
-            self.room.cleaner = asyncio.create_task(self._clean_room())
-        if self._room_id != "JupyterLab:globalAwareness":
-            self._emit_awareness_event(self.current_user.username, "leave")
+        if hasattr(self, "room"):
```

Comment:
I'd rather we not use `hasattr` and implicit attributes here. 

Instead, let's initialize the `room` and `_room_id` attributes in the `initialize(...)` method, even if we set them to `None` initially: https://github.com/jupyterlab/jupyter-collaboration/blob/926c4d000b26cb677aca93cf3868f6a4c11f0e8b/projects/jupyter-server-ydoc/jupyter_server_ydoc/handlers.py#L120-L136

I think that's the way to do it in tornado handlers.